play-akka-http-ws-standalone/src/main/java/play/libs/ws/akkahttp/StandaloneAkkaHttpWSResponse.java

```diff
@@ -0,0 +1,211 @@
+   * own {@link BodyReadable} types:
+   * <p>
+   * <pre>
+   * {@code public class MyClass {
+   *   private BodyReadable<Foo, StandaloneWSResponse> fooReadable = (response) -> new Foo();
+   *
+   *   public void readAsFoo(StandaloneWSResponse response) {
+   *       Foo foo = response.getBody(fooReadable);
+   *   }
+   * }
+   * }
+   * </pre>
+   * <p>
+   * or use {@code play.libs.ws.ahc.DefaultResponseReadables}
+   * for the built-ins:
+   * <p>
+   * <pre>
+   * {@code public class MyClass implements DefaultResponseReadables {
+   *     public void readAsString(StandaloneWSResponse response) {
+   *         String getBody = response.getBody(string());
+   *     }
+   *
+   *     public void readAsJson(StandaloneWSResponse response) {
+   *         JsonNode json = response.getBody(json());
+   *     }
+   * }
+   * }
+   * </pre>
+   *
+   * @param readable the readable to convert the response to a T
+   * @return the response getBody transformed into an instance of T
+   */
+  @Override
+  public <T> T getBody(BodyReadable<T> readable) {
+    return readable.apply(this);
+  }
+
+  /**
+   * The response body decoded as String, using a simple algorithm to guess the encoding.
+   * <p>
+   * This decodes the body to a string representation based on the following algorithm:
+   * <p>
+   * 1. Look for a "charset" parameter on the Content-Type. If it exists, set `charset` to its value and goto step 3.
+   * 2. If the Content-Type is of type "text", set $charset to "ISO-8859-1"; else set `charset` to "UTF-8".
+   * 3. Decode the raw bytes of the body using `charset`.
+   * <p>
+   * Note that this does not take into account any special cases for specific content types. For example, for
+   * application/json, we do not support encoding autodetection and will trust the charset parameter if provided.
+   *
+   * @return the response body parsed as a String using the above algorithm.
+   */
+  @Override
+  public String getBody() {
+    return getBodyAsBytes().utf8String();
+  }
+
+  @Override
+  public ByteString getBodyAsBytes() {
+    try {
+      // FIXME JAVA API no Unmarshalling Java API in Akka Http
+      return getStrictResponse().entity().getDataBytes()
+        .runWith(Sink.fold(ByteString.empty(), (b1, b2) -> b1.concat(b2)), mat)
+        .toCompletableFuture()
+        .get(UNMARSHAL_TIMEOUT.toNanos(), TimeUnit.NANOSECONDS);
+    }
+    catch (Exception ex) {
+      throw new RuntimeException(ex);
+    }
+  }
+
+  @Override
+  public Source<ByteString, ?> getBodyAsSource() {
+    return response.entity().getDataBytes();
+  }
+
+  private synchronized HttpResponse getStrictResponse() {
+    if (strictResponse != null) {
+      return strictResponse;
+    }
+    else {
+      try {
+        final ResponseEntity strictEntity = response.entity()
+          .toStrict(UNMARSHAL_TIMEOUT.toMillis(), mat)
+          .toCompletableFuture()
+          .get(UNMARSHAL_TIMEOUT.toNanos(), TimeUnit.NANOSECONDS);
```

Comment:
So this is also different from AHC implementation, right? I don't think we have a timeout when reading strict entities in AHC. Maybe we need a more descriptive exception message below to instruct users that maybe they need to use `getBodyAsSource`.